### ISSUE TITLE
unexpected dependencies 'jest-diff',   'jest-matcher-utils'

### ISSUE THREAD

**Issue by echoinsane** [just-jeb/jest-marbles#264]
Hi!

After i removed **jest-cli**, i started to getting errors:

`Cannot find module 'jest-diff' from 'jest-marbles.js'`
`Cannot find module 'jest-matcher-utils' from 'jest-marbles.js'`

_package.json:_ 
```
"devDependencies": {
        "@angular-builders/jest": "^8.2.0",
        "@angular-devkit/build-angular": "^0.803.18",
        "@angular/cli": "^8.3.18",
        "@angular/compiler-cli": "^8.2.13",
        "@angular/language-service": "^8.2.13",
        "@types/accounting": "0.3.29",
        "@types/jest": "^24.0.22",
        "@types/lodash": "4.14.121",
        "@types/node": "~11.9.6",
        "@types/selenium-webdriver": "3.0.15",
        "@types/source-map": "0.5.7",
        "accounting": "0.3.2",
        "angular2-template-loader": "^0.6.2",
        "canvas-toBlob": "1.0.0",
        "cpy-cli": "1.0.1",
        "expect": "^24.9.0",
        "husky": "0.14.3",
        "jest": "^23.6.0",
        "jest-marbles": "^2.5.0",
        "npm-run-all": "4.0.1",
        "prettier": "1.16.4",
        "pretty-quick": "1.10.0",
        "raw-loader": "^0.5.1",
        "rimraf": "2.6.2",
        "stylelint-teamcity-reporter": "^1.0.0",
        "text-mask-addons": "3.7.1",
        "text-mask-core": "5.0.1",
        "ts-jest": "^24.1.0",
        "ts-mockito": "^2.5.0",
        "tslint-teamcity-reporter": "^2.0.0",
        "typescript": "^3.5.3",
        "webpack-bundle-analyzer": "3.3.2"
    },
```

**Comment by nezed** [comment]
Those packages are imported in https://github.com/just-jeb/jest-marbles/blob/475a3cb0dfa7e5f1ef056c5b7e3fde4279b8cc69/src/jest/custom-matchers.ts#L2-L3
but isn't listed in `jest-marbles` dependencies or even peerDependencies.

Looks like there is nothing to guarantee by `jest-marbles` that `jest-diff` and `jest-matcher-utils` are installed and accessible.

Example of dep-tree with `jest-cli` installed:
```
$ npm ls jest-diff
my-project
├─┬ expect@24.9.0
│ └─┬ jest-matcher-utils@24.9.0
│   └── jest-diff@24.9.0
└─┬ jest-cli@24.8.0
  └─┬ @jest/core@24.9.0
    └─┬ jest-snapshot@24.9.0
      ├── jest-diff@24.9.0
      └─┬ jest-matcher-utils@24.9.0
        └── jest-diff@24.9.0  deduped
```
`jest-diff` was hoisted by npm and becomes accessible to `jest-marbles`

Without `jest-cli` or any others depending on it:
```
$ npm ls jest-diff
my-project
└─┬ expect@24.9.0
  └─┬ jest-matcher-utils@24.9.0
    └── jest-diff@24.9.0
```

**Comment by just-jeb** [comment]
I guess you're right, need to list them in the dependencies.

**Comment by just-jeb** [comment]
But if I remember correcly `jest-cli` is `jest` dependency so when you install `jest` you also install `jest-cli`. Since `jest-marbles` is extension to Jest framework it's a valid assumption that `jest` is installed. Why removing `jest-cli`, what is your use case exactly?

**Comment by nezed** [comment]
You right, but i'll describe you why it will not help.

We've had `jest-cli` in our `devDependencies`, so dep-tree looked like:
```
my-project
├── jest-marbles
├─┬ jest
│ └─┬ jest-cli   deduped
│   └── jest-diff  deduped
└─┬ jest-cli
  └─┬ jest-snapshot
    ├── jest-diff  deduped
    └─┬ jest-matcher-utils
      └── jest-diff  deduped
```
Because `jest-diff` was deduped by npm, it was hoisted to `./node_modules/jest-diff/`, and becomes accessible to any dependency, even for `jest-marbles` (it tries to require `./node_modules/jest-diff/`, then `../node_modules/jest-diff/`, then `../../node_modules/jest-diff/` etc)

When we've removed `jest-cli` from our devDeps there is not reasons to dedupe, so dep-tree looks  this way:
```
my-project
├── jest-marbles
└─┬ expect
  └─┬ jest-matcher-utils
    └── jest-diff
```
As you can see, `jest-diff` is presented, but can not be required from `jest-marbles`

**Comment by just-jeb** [comment]
Fixed in `2.5.1`.
